From 384d2624ef72678004079c7bd9ef70b1ee594357 Mon Sep 17 00:00:00 2001
From: Xiaolei Wang <xiaolei.wang@windriver.com>
Date: Thu, 4 Jun 2020 16:01:33 +0800
Subject: [PATCH] clk: ti: am33xx: fix a warning when pruss module is loaded

Register access failed when the clock is enabled, so a warning will
appear when disable, AM3_PRUSS_CLKCTR should be used to replace
AM3_PRUSS_OCP_PRUSS_CLKCTRL

Fix warning for:
WARNING: CPU: 0 PID: 189 at drivers/clk/clk.c:946
clk_core_disable+0xf8/0x284
pruss-ocp-clkctrl:0000:0 already disabled
Modules linked in: pruss_soc_bus  wlcore_sdio  snd_soc_davinci_mcasp
snd_soc_ti_edma snd_soc_ti_sdma snd_soc_tlv320aic3x omap_wdt
wkup_m3_rproc snd_soc_core remoteproc musb_am335x snd_pcm_dmaen4
CPU: 0 PID: 189 Comm: systemd-udevd Not tainted 5.2.41-yocto-standard #1
Hardware name: Generic AM33XX (Flattened Device Tree)
[<c0110cec>] (unwind_backtrace) from [<c010d2d0>] (show_stack+0x18/0x1c)
[<c010d2d0>] (show_stack) from [<c0131b6c>] (__warn.part.0+0xc4/0xdc)
[<c0131b6c>] (__warn.part.0) from [<c0131c34>]
(warn_slowpath_fmt+0xb0/0xdc)
[<c0131c34>] (warn_slowpath_fmt) from [<c06a42f4>]
(clk_core_disable+0xf8/0x284)
[<c06a42f4>] (clk_core_disable) from [<c06a4864>]
(clk_core_disable_lock+0x20/0x2c)
[<c06a4864>] (clk_core_disable_lock) from [<c0121cf4>]
(_disable_clocks+0x20/0xa0)
[<c0121cf4>] (_disable_clocks) from [<c0121e34>]
(_deassert_hardreset+0xc0/0x168)
[<c0121e34>] (_deassert_hardreset) from [<c0124ab8>]
(omap_hwmod_deassert_hardreset+0x40/0x8c)
[<c0124ab8>] (omap_hwmod_deassert_hardreset) from [<c0125898>]
(omap_device_deassert_hardreset+0x3c/0x58)
[<c0125898>] (omap_device_deassert_hardreset) from [<bf1021a0>]
(pruss_soc_bus_probe+0xbc/0x218 [pruss_soc_bus])
[<bf1021a0>] (pruss_soc_bus_probe [pruss_soc_bus]) from [<c077e2c0>]
(platform_drv_probe+0x50/0xa0)
[<c077e2c0>] (platform_drv_probe) from [<c077bfe0>]
(really_probe+0xf8/0x2d4)
[<c077bfe0>] (really_probe) from [<c077c354>]
(driver_probe_device+0x68/0x174)
[<c077c354>] (driver_probe_device) from [<c077c628>]
(device_driver_attach+0x60/0x68)
[<c077c628>] (device_driver_attach) from [<c077c690>]
(__driver_attach+0x60/0xd4)
[<c077c690>] (__driver_attach) from [<c077a0e4>]
(bus_for_each_dev+0x80/0xc8)
[<c077a0e4>] (bus_for_each_dev) from [<c077b3e8>]
(bus_add_driver+0x168/0x1ec)
[<c077b3e8>] (bus_add_driver) from [<c077d234>]
(driver_register+0x7c/0x110)
[<c077d234>] (driver_register) from [<c0103578>]
(do_one_initcall+0x60/0x260)
[<c0103578>] (do_one_initcall) from [<c01a6bc8>]
(do_init_module+0x68/0x248)
[<c01a6bc8>] (do_init_module) from [<c01a8da8>]
(load_module+0x1f74/0x23a8)
[<c01a8da8>] (load_module) from [<c01a945c>]
(sys_finit_module+0xd8/0xf0)
[<c01a945c>] (sys_finit_module) from [<c0101000>]
(ret_fast_syscall+0x0/0x54)
Exception stack(0xcd41ffa8 to 0xcd41fff0)
ffa0: 00000000 00000000 00000006 b6ee2520 00000000 00020000
ffc0: 00000000 00000000 00000000 0000017b 00000001 00000000 0053c99b
00000000
ffe0: bed47670 bed47660 b6edcaa9 b6dd0f52

Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 drivers/clk/ti/clk-33xx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/ti/clk-33xx.c b/drivers/clk/ti/clk-33xx.c
index a360d3109555..18c9f856f1fe 100644
--- a/drivers/clk/ti/clk-33xx.c
+++ b/drivers/clk/ti/clk-33xx.c
@@ -107,7 +107,7 @@ static const struct omap_clkctrl_reg_data am3_l4hs_clkctrl_regs[] __initconst =
 };
 
 static const struct omap_clkctrl_reg_data am3_pruss_ocp_clkctrl_regs[] __initconst = {
-	{ AM3_PRUSS_OCP_PRUSS_CLKCTRL, NULL, CLKF_SW_SUP, "pruss_ocp_gclk" },
+	{ AM3_PRUSS_CLKCTRL, NULL, CLKF_SW_SUP, "pruss_ocp_gclk" },
 	{ 0 },
 };
 
-- 
2.17.1

